python/pyarrow/table.pxi

```diff
@@ -565,6 +565,10 @@ cdef class RecordBatch:
         self.batch = NULL
         self._schema = None
 
+    def __init__(self):
+        raise RuntimeError("Do not call RecordBatch's constructor directly, "
+                           "use one of the `from_*` methods instead.")
```

Comment:
Nit: they are not methods but functions. Or perhaps there are both, so say "methods and functions".